src/main/java/com/linecorp/armeria/common/util/Exceptions.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+import java.io.IOException;
+import java.nio.channels.ClosedChannelException;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+
+import com.linecorp.armeria.client.ClosedSessionException;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelException;
+import io.netty.handler.codec.http2.Http2Exception;
+
+public final class Exceptions {
+
+    private static final Pattern IGNORABLE_SOCKET_ERROR_MESSAGE = Pattern.compile(
+            "(?:connection.*(?:reset|closed|abort|broken)|broken.*pipe)", Pattern.CASE_INSENSITIVE);
+
+    private static final Pattern IGNORABLE_HTTP2_ERROR_MESSAGE = Pattern.compile(
+            "(?:stream closed)", Pattern.CASE_INSENSITIVE);
+
+
+    public static void log(Logger logger, Channel ch, Throwable cause) {
+        if (!logger.isWarnEnabled()) {
+            return;
+        }
+
+        if (needsAttention(cause)) {
+            logger.warn("{} Unexpected exception:", ch, cause);
+        }
+    }
+
+    public static boolean needsAttention(Throwable cause) {
```

Comment:
Will rename as suggested (`isExpected()` and `logIfUnexpected()`) and add Javadoc I forgot.